vllm/spec_decode/multi_step_worker.py

```diff
@@ -24,14 +29,67 @@ class MultiStepWorker(Worker, ProposerWorkerBase):
     requires more thought for MultiStepWorker support.
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, draft_ranks: Optional[List[int]] = None, **kwargs):
```

Comment:
Is there any reason the PR combines draft-tensor-parallel-degree logic and multi-step-worker logic into the same class? It seems to me the code could be simpler if there is a separate `SingleTpWorker` within which a `MultiStepWorker` is composed.

Reply:
We combined them because 'SingleTpWorker' is always used for 'MultiStepWorker'.

I also agree with you, so I can revert the code if you think that's the better option.
I'll add this work as the last single commit.